components/ai_chat/resources/page/components/site_title/index.tsx

```diff
@@ -58,16 +58,18 @@ function SiteTitle () {
 
   const handlePageContentDisconnect = () => {
     getPageHandlerInstance().pageHandler.disconnectPageContents()
+     // Refetch the preference to ensure the user sees the prompt without waiting for a new runtime.
+    context.maybeShowPremiumPrompt()
```

Comment:
This is a side effect. If premium prompt is dependent on whether there is site info or not, can the context just check that?